kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnector.java

```diff
@@ -85,4 +92,38 @@ public ConfigDef config() {
     return S3SinkConnectorConfig.getConfig();
   }
 
+  public boolean checkBucketExists(String bucketName) {
+    try {
+      final AmazonS3 bucketCheckHelper = AmazonS3ClientBuilder.defaultClient();
+      return bucketCheckHelper.doesBucketExist(bucketName);
+    } catch (SdkClientException e) {
+      return false;
+    }
+  }
+
+  @Override
+  public Config validate(final Map<String, String> connectorConfigs) {
```

Comment:
Had to override validate() method instead of adding a validator to the bucket-name ConfigDef due to:
- Adding a validator to a property requires a default value.
- Default value of empty string will fail its own validator.
- Other valid bucket names do not make much sense since no bucket is always guaranteed to exist.

Reply:
you don't need to override the `validate()` method. You can write your own custom validator that implements custom validation logic like [here](https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java#L629-L660) and thus can be implemented in the configs

Reply:
More critically, your assumption "Adding a validator to a property requires a default value" is false. 
You can use [ConfigDef.NO_DEFAULT_VALUE](https://kafka.apache.org/0110/javadoc/org/apache/kafka/common/config/ConfigDef.html#NO_DEFAULT_VALUE) to indicate that the user must always provide a value, even if you're using one of the define() signatures that includes a default value as one of the arguments.